fe/fe-core/src/main/java/org/apache/doris/nereids/rules/exploration/mv/AbstractMaterializedViewAggregateRule.java

```diff
@@ -230,57 +230,56 @@ protected Plan rewriteQueryByView(MatchMode matchMode,
-            } else {
-                // if group by expression, try to rewrite group by expression
-                Expression queryGroupShuttledExpr = ExpressionUtils.shuttleExpressionWithLineage(
-                        topExpression, queryTopPlan, queryStructInfo.getTableBitSet());
-                AggregateExpressionRewriteContext context = new AggregateExpressionRewriteContext(true,
-                        mvExprToMvScanExprQueryBased, queryTopPlan, queryStructInfo.getTableBitSet());
-                // group by expression maybe group by a + b, so we need expression rewriter
-                Expression rewrittenGroupByExpression = queryGroupShuttledExpr.accept(AGGREGATE_EXPRESSION_REWRITER,
-                        context);
-                if (!context.isValid()) {
-                    // group expr can not rewrite by view
-                    materializationContext.recordFailReason(queryStructInfo,
-                            "View dimensions doesn't not cover the query dimensions",
-                            () -> String.format("mvExprToMvScanExprQueryBased is %s,\n queryGroupShuttledExpr is %s",
-                                    mvExprToMvScanExprQueryBased, queryGroupShuttledExpr));
-                    return null;
-                }
-                NamedExpression groupByExpression = rewrittenGroupByExpression instanceof NamedExpression
-                        ? (NamedExpression) rewrittenGroupByExpression : new Alias(rewrittenGroupByExpression);
-                finalOutputExpressions.add(groupByExpression);
-                finalGroupExpressions.add(groupByExpression);
+            Expression queryFunctionShuttled = ExpressionUtils.shuttleExpressionWithLineage(
+                    topExpression,
+                    queryTopPlan,
+                    queryStructInfo.getTableBitSet());
+            AggregateExpressionRewriteContext context = new AggregateExpressionRewriteContext(
+                    false, mvExprToMvScanExprQueryBased, queryTopPlan, queryStructInfo.getTableBitSet());
+            // queryFunctionShuttled maybe sum(column) + count(*), so need to use expression rewriter
+            Expression rollupedExpression = queryFunctionShuttled.accept(AGGREGATE_EXPRESSION_REWRITER,
+                    context);
+            if (!context.isValid()) {
+                materializationContext.recordFailReason(queryStructInfo,
+                        "Query function roll up fail",
+                        () -> String.format("queryFunctionShuttled = %s,\n mvExprToMvScanExprQueryBased = %s",
+                                queryFunctionShuttled, mvExprToMvScanExprQueryBased));
+                return null;
             }
+            finalOutputExpressions.add(new Alias(rollupedExpression));
         }
         // add project to guarantee group by column ref is slot reference,
         // this is necessary because physical createHash will need slotReference later
```

Comment:
Maybe add the following logic here is better, WDYT? 
// if top plan doesn't use group by but bottom agg has group by, should also check
        if (!queryGroupByExpressions.isEmpty() && queryTopPlanGroupBySet.isEmpty()) {
            // todo check as code in line 263 to 278
        }

Reply:
I tried here but caused duplicated code about `rewrittenGroupByExpression`, so I just add the compensate groupby to `queryExpressions`.

Reply:
Duplicated code can extract a common method and call the method in different place.
`queryExpressions` means the expression query used readly, if you add compensate groupby to `queryExpressions`, this may change the `queryExpressions` original meaning.